File: ProjectStart/ProjectStart.WebApp/ClientApp/Shared/AppModels/BaseEntity.ts
import { HistoryEntity } from 'shared/AppModels/HistoryEntity';
/**
 * Interface for: ProjectStart.Entity.Common.BaseEntity
 */
export interface IBaseEntity  {
    id: string;
    name: string;
    history: HistoryEntity;
}
/**
 * Base view model for ProjectStart.Entity.Common.BaseEntity
 */
export class BaseEntity  implements IBaseEntity {
    id: string;
    name: string;
    history: HistoryEntity;
}

File: ProjectStart/ProjectStart.Entity/Common/BaseEntity.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ProjectStart.Entity.Common
{
    public abstract class BaseEntity
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity), ForeignKey("History")]
        public string Id { get; set; }
        [Required]
        public string Name { get; set; }
        public HistoryEntity History { get; set; } = new HistoryEntity();
    }
}
